contrib/opencensus-ext-azure/opencensus/ext/azure/trace_exporter/__init__.py

```diff
@@ -87,32 +92,45 @@ def span_data_to_envelope(self, sd):
                 data.success = (
                     status_code >= 200 and status_code <= 399
                 )
+            elif sd.status.code == 0:
+                data.success = True
         else:
             envelope.name = \
                 'Microsoft.ApplicationInsights.RemoteDependency'
             data = RemoteDependency(
                 name=sd.name,  # TODO
-                id='|{}.{}.'.format(sd.context.trace_id, sd.span_id),
-                resultCode='0',  # TODO
+                id='{}'.format(sd.span_id),
+                resultCode=str(sd.status.code),
                 duration=utils.timestamp_to_duration(
                     sd.start_time,
                     sd.end_time,
                 ),
-                success=True,  # TODO
+                success=False,  # Modify based off attributes or status
                 properties={},
             )
             envelope.data = Data(
                 baseData=data,
                 baseType='RemoteDependencyData',
             )
             if sd.span_kind == SpanKind.CLIENT:
-                data.type = 'HTTP'  # TODO
+                data.type = sd.attributes.get('component')
                 if 'http.url' in sd.attributes:
                     url = sd.attributes['http.url']
                     # TODO: error handling, probably put scheme as well
-                    data.name = utils.url_to_dependency_name(url)
+                    data.data = url
+                    parse_url = urlparse(url)
+                    # target matches authority (host:port)
+                    data.target = parse_url.netloc
+                    if 'http.method' in sd.attributes:
+                        # name is METHOD/path
+                        data.name = sd.attributes['http.method'] \
+                            + "/" + parse_url.path
                 if 'http.status_code' in sd.attributes:
-                    data.resultCode = str(sd.attributes['http.status_code'])
+                    status_code = sd.attributes["http.status_code"]
+                    data.resultCode = str(status_code)
+                    data.success = 200 <= status_code < 400
+                elif sd.status.code == 0:
+                    data.success = True
```

Comment:
what if it's not SERVER or CLIENT span (Unspecified), we create Dependency and we should still set status